config/global.go

```diff
@@ -97,6 +99,19 @@ func GetDaemonProfileCPUDuration() int64 {
 }
 
 func ProcessConfigurations(c *SnapshotterConfig) error {
+	for {
+		realPath, err := mount.NormalizePath(c.Root)
+		if err != nil {
+			if !os.IsNotExist(errors.Cause(err)) {
+				return errors.Wrapf(err, "Root path invalid.")
+			}
+			if err := os.MkdirAll(c.Root, 0755); err != nil {
+				return errors.Wrapf(err, "create root dir %s", c.Root)
+			}
+		}
+		c.Root = realPath
+		break
```

Comment:
Can we restrict the permissions to `0700`? 
A more straightforward way would be to create the root dir (relative)  and then normalize the relative path. Then reassign it to `c.Root`. It is safe to remove the current code of creating root dir then. Furthermore, we don't have to check the target dir's existence before `MkdirAll`

Moreover, why a relative path is set for a long-lived background service, which makes it a little vague?

Reply:
For users it doesn't matter, but for developers, I just want to simply use my local directory.

Reply:
Or if the user configures a relative path, it will directly return the startup failure, otherwise it will be very difficult to debug when the containerd reports an error.